### ISSUE TITLE
Unable to download backups due to "Not authenticated" error

### ISSUE THREAD

**Issue by evulhotdog** [hay-kot/mealie#279]
**Describe the bug**
When attempting to download any backups on 0.4.0, it says I'm unable to download it with the following json payload:\
```
{"detail":"Not authenticated"}
```

**Steps To Reproduce**
Please be specific!
1. Navigate to mealie in browser
2. Log in to mealie
3. Go to settings
4. Go to backups
5. Select a backup
6. Click download
7. It redirects you to `https://foo.bar/api/backups/pre-0.4.1-backup_2021-Apr-09.zip/download` which provides the json: `{"detail":"Not authenticated"}`

**Sample Code**

**Expected behavior**
My browser is prompted to download a file.

**Actual Behavior**
It redirects you to `https://foo.bar/api/backups/pre-0.4.1-backup_2021-Apr-09.zip/download` which provides the json: `{"detail":"Not authenticated"}`

**Screenshots**

![image](https://user-images.githubusercontent.com/365456/114221016-b93f0500-993a-11eb-8240-a63d9c4ec16f.png)
![image](https://user-images.githubusercontent.com/365456/114220957-a2001780-993a-11eb-9074-6f6141929359.png)
![image](https://user-images.githubusercontent.com/365456/114221032-be9c4f80-993a-11eb-95e0-10607e4a8f9d.png)

**Device Information (please complete the following information):**
 - OS: host is Ubuntu
 - Deployment: Docker latest
 - Browser: Chrome 89.x
 - Version:  0.4.0

**Additional context**
This worked in previous versions of mealie.

**Comment by hay-kot** [comment]
This has been fixed in v0.4.1 and v0.4.1a

**Comment by evulhotdog** [comment]
@hay-kot is there risk of upgrading and losing the recipes on 0.4.1x? 
How do you recommend I go about upgrading?

**Comment by hay-kot** [comment]
> @hay-kot is there risk of upgrading and losing the recipes on 0.4.1x?
> How do you recommend I go about upgrading?

There are no database changes in this release so there shouldn't be any issues related to databases. The 'major' change in this release is image minification where on-startup the images are migrated to a new location and are minified (keeping the original image) and split into 3 sizes, original, mini, and tiny. I have successfully migrated 3 instances without error using the v0.4.1a release. 

Note that the v0.4.1 release had a bug that prevented the startup scripts to run so make sure you pull either the latest or the v0.4.1a tags.

That said, there is no reason Mealie should every delete a backup unless you do it from the UI so if you create a backup it shouldn't be deleted on upgrade. The only reason it may be deleted that I can think of is if your docker volume is not persistent. 

You can find more info [Here](https://hay-kot.github.io/mealie/changelog/v0.4.1/)

**Comment by evulhotdog** [comment]
First and foremost, thank you for the help.

My concern was that there are not database migrations that happen automatically, so I always backup before any update, manually. Then from there, I reimport to ensure things are fresh. Because I can't backup here, and there were new recipes I've added, I was concerned about losing them, in the event there was a catastrophic change. Either way, I can back up the volume and revert if something does happen. Plus since you mentioned it's a minor change with no db changes, this is a lot less risky.

**Comment by hay-kot** [comment]
I 100% agree with everything you've mentioned and they are valid concerns. Losing data is terrifying. I've nuked my whole self-hosted setup before without backups and it sucks. 

I don't believe I've done the best job at identifying what is required to update a Mealie instance and what impacts it may have. I've now added a big red banner that indicates breaking changes in change logs now as well as indicated the database version and app version. If the database version is the same, no database changes occur. 

Perhaps that is not as clear as I had thought. I'll consider adding a 'upgrade process' section to the change log to identify any steps that may be necessary. On the bright side, I think we're nearing a steady database version so hopefully migrations are in the near future...

Thanks for closing the issue and being a user!
